src/main/java/dev/gigaherz/enderrift/rift/RiftStructure.java

```diff
@@ -182,12 +183,14 @@ private static void buildStructure(Level world, BlockPos pos, ItemStack itemStac
 
         if (tagCompound != null && tagCompound.contains("RiftId"))
         {
-            rift.assemble(tagCompound.getInt("RiftId"));
-        }
-        else
-        {
-            rift.assemble(RiftStorage.get(world).getNextRiftId());
+            UUID id = tagCompound.getUUID("RiftId");
```

Comment:
This might need migration as well as the Rift Core Item could still contain an old rift id

Reply:
I'll add that later

Reply:
[Used verifyTagAfterLoad](https://github.com/gigaherz/Ender-Rift/pull/56/commits/9872c64fd9350429e87f0b93aeea34bf7763d8c5#diff-4aea2f72b433984cdd51355dbbc472b0349cc5a1f90eaee263fa89fb0847c9d9R42) to migrate